src/main/java/tasks/Deadline.java

```diff
@@ -0,0 +1,29 @@
+package tasks;
+
+import utils.DateParser;
+
+import java.time.LocalDate;
+
+public class Deadline extends Task {
+
+    private final LocalDate localDate;
+
+    public Deadline(String name, String dateStr) {
+        super(name);
+        localDate = DateParser.stringToDate(dateStr);
+    }
+
+    @Override
+    public String toString() {
+        return "[D] " + super.toString() + " (by: " + localDate.toString() + ")";
+    }
+
+    @Override
+    public String toDataString() {
+        return String.format("[D] | %d | %s | %s",
+                isMarked() ? 1 : 0,
```

Comment:
Perhaps you could look into replacing the 1 and 0 in the conditional statement with a named constant? It could be considered a "magic number".